dotnet/src/Azure.Iot.Operations.Protocol/RPC/CommandExecutor.cs

```diff
@@ -475,7 +475,12 @@ private async Task<MqttApplicationMessage> GenerateResponseAsync(
             string timestamp = await _applicationContext.ApplicationHlc.UpdateNowAsync();
             message.AddUserProperty(AkriSystemProperties.Timestamp, timestamp);
 
-            EnsureCloudEventTypeAndContentType(metadata, _msAioRpcResponse, payloadContext?.ContentType);
+            if (metadata?.CloudEvent != null)
+            {
+                // Set Type and DataContentType for RPC response CloudEvents
+                metadata.CloudEvent.Type = _msAioRpcResponse;
```

Comment:
this should only be set if the customer didn't set a type (it should be a default, not an override)